docs/manual/developer_guide.adoc

```diff
@@ -73,11 +76,22 @@ image::Patching-Pull_Requests.png[align="left"]
 image::Patching-Create_Pull.png[align="left"]
 
 4) You will be requested to input a patch title and description. Be concise, but thorough enough for a community member to understand logic behind your changes. Paste into the description field testing evidence (e.g. running testoval.py on any submitted OVAL, or before/after for remediation scripts).
+
+If you work on a feature or a bugfix that has an associated issue, either
+
+- assign yourself to the issue (if you have rights), or
+- mention the issue number in the pull request.
+
+This will improve the odds that multiple people won't work on a single issue without being aware of each other's work.
 After completing the form, select "Send pull request":
 
 image::Patching-Send_Pull_Request.png[align="left"]
 
-5) A community member will review your patch. They will either merge the patch upstream, indicate additional changes/documentation needed, or decline the patch. You'll automatically be notified via EMail of any status updates.
+5) Keep historical records clean - don't use git history-altering magic in your branch during your work on the pull request.
+If you e.g. squash commits, the pull request page will be broken - if you made some mistakes, got feedback and corrected those mistakes based on the feedback, nobody will be able to learn from your pull request, because commits introducing mistakes will disappear and comments of reviewers therefore won't make sense.
+You can always squash commits while merging the pull request, or you can close a complicated pull request, create a new one (in another branch) with streamlined commits and reference the old PR in the streamlined one, so it is possible to backtrack what went on.
```

Comment:
Squash unnecessary commits when merging the pull request, or close a complicated pull request and create a new one (in another branch) with streamlined commits. Reference the old PR in the new streamlined pull request so it is possible to backtrack what went on.